OpenSim/Common/Test/testComponentInterface.cpp

```diff
@@ -897,52 +897,15 @@ void testListSockets() {
 
 void testComponentPathNames()
 {
-    Foo foo;
-    Bar bar;
+    // Build using real components and assemble them 
+    // into a tree and test the path names that are 
+    // generated on the fly.
     TheWorld top;
+    TheWorld otherTop;
 
-    // These are not valid component names
-    // Only using for testing as surrogates for path names
-    // which are computed by the component. Just testing
-    // the relative path name facility here.
     top.setName("Top");
-    foo.setName("A/B/C/D");
-    bar.setName("A/B/E");
-
-    std::string fooWrtBar = foo.getRelativePathName(bar);
-    ASSERT(fooWrtBar == "../C/D"); // "/A/B/" as common
-
-    std::string barWrtFoo= bar.getRelativePathName(foo);
-    ASSERT(barWrtFoo == "../../E"); // "/A/B/" as common
-
-    // null case foo wrt foo
-    std::string fooWrtFoo = foo.getRelativePathName(foo);
-    ASSERT(fooWrtFoo == "");
-
-    std::string topAbsPath = top.getAbsolutePathName();
-    std::string fooWrtTop = foo.getRelativePathName(top);
-    ASSERT(fooWrtTop == "../A/B/C/D");
-
-    std::string topWrtFoo = top.getRelativePathName(foo);
-    ASSERT(topWrtFoo== "../../../../Top");
-
-    foo.setName("World/Foo");
-    bar.setName("World3/bar2");
-    fooWrtBar = foo.getRelativePathName(bar);
-    ASSERT(fooWrtBar == "../../World/Foo");
-
-    foo.setName("World3/bar2/foo1");
-    fooWrtBar = foo.getRelativePathName(bar);
-    ASSERT(fooWrtBar == "foo1");
+    otherTop.setName("OtherTop");
 
-    bar.setName("LegWithConstrainedFoot/footConstraint");
-    foo.setName("LegWithConstrainedFoot/foot");
-    barWrtFoo = bar.getRelativePathName(foo);
-    ASSERT(barWrtFoo == "../footConstraint");
```

Comment:
I didn't look too carefully, but have some test cases been removed? If so, could you explain why?

Reply:
These tests looked like they were redundant with other tests, so I trimmed them down. From what I could tell, they may have added 1) testing with longer names and 2) testing after `setName()` was invoked on a `Component` that already existed. If that's important to go back in (or if I missed the reason for these tests), I'm happy to do re-create these tests so that they'll work with the interface changes.

Reply:
Okay sounds fine.